hardhat.config.ts

```diff
@@ -0,0 +1,142 @@
+import '@nomiclabs/hardhat-ethers'
+import '@nomiclabs/hardhat-solhint'
+import '@nomiclabs/hardhat-truffle5'
+import '@nomiclabs/hardhat-waffle'
+import dotenv from 'dotenv'
+import { existsSync } from 'fs'
+import fs from 'fs/promises'
+import 'hardhat-abi-exporter'
+import 'hardhat-deploy'
+import 'hardhat-gas-reporter'
+import { HardhatUserConfig, task } from 'hardhat/config'
+
+const archivedDeploymentPath = './deployments/archive'
+
+// Load environment variables from .env file. Suppress warnings using silent
+// if this file is missing. dotenv will never modify any environment variables
+// that have already been set.
+// https://github.com/motdotla/dotenv
+dotenv.config({ debug: false })
+
+task('accounts', 'Prints the list of accounts', async (_, hre) => {
+  const accounts = await hre.ethers.getSigners()
+
+  for (const account of accounts) {
+    console.log(account.address)
+  }
+})
+
+task('save', 'Saves a specified contract as a deployed contract')
+  .addPositionalParam('contract', 'The contract to save')
+  .addPositionalParam('block', 'The block number the contract was deployed at')
+  .setAction(
+    async ({ contract, block }: { contract: string; block: string }, hre) => {
+      const network = hre.network.name
+      const artifact = await hre.deployments.getArtifact(contract)
+
+      const archiveName = `${contract}_${network}_${block}`
+      const archivePath = `${archivedDeploymentPath}/${archiveName}.sol`
+
+      if (existsSync(archivePath)) {
+        throw new Error('Archive already exists')
+      }
+
+      let newArtifact = artifact
+      newArtifact.contractName = archiveName
+      newArtifact.sourceName = archivePath.substring(2)
+
```

Comment:
Maybe log the commit/tree hash here too?

Reply:
Could definitely do this, although the artifacts follow the hardhat-deploy types so adding it would be not following their spec

Reply:
If it will let you extend the spec, it would be good to have that data.